code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm

```diff
@@ -185,3 +245,160 @@
+	return ..()
+
+/datum/action/xeno_action/activable/rage/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return
+
+	var/mob/living/carbon/xenomorph/ravager/rager = owner
+
+	var/rage_health_threshold = rager.maxHealth * 0.5 //Need to be at 50% of max hp or lower to rage
+	if(rager.health > rage_health_threshold)
+		if(!silent)
+			to_chat(rager, "<span class='xenodanger'>Our health isn't low enough to rage! We must take [rager.health - rage_health_threshold] more damage!</span>")
+		return FALSE
+
+
+/datum/action/xeno_action/activable/rage/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/ravager/X = owner
+
+	rage_power = (1-(X.health/X.maxHealth)) * 0.5 //Calculate the power of our rage; scales with difference between current and max HP
+
+	if(X.health < 0) //Double dip on the negative HP; gain + 0.005 rage power per point of negative HP
+		rage_power += X.health * -0.005
+
+	rage_power = min(1, rage_power) //Cap rage power so that we don't get way too insane.
+
+	var/rage_power_radius = CEILING(rage_power * 6, 1) //Define radius of the SFX
+
+	X.visible_message("<span class='danger'>\The [X] becomes frenzied, bellowing with a shuddering roar!</span>", \
+	"<span class='highdanger'>We bellow as our fury overtakes us! RIP AND TEAR!</span>")
+	X.do_jitter_animation(1000)
+
+
+	//Roar SFX; volume scales with rage
+	playsound(X.loc, 'sound/voice/alien_roar2.ogg', clamp(100 * rage_power, 25, 80), 0)
+
+	var/bonus_duration
+	if(rage_power > 0.5) //If we're super pissed it's time to get crazy
+		var/datum/action/xeno_action/charge = X.actions_by_path[/datum/action/xeno_action/activable/charge]
+		var/datum/action/xeno_action/ravage = X.actions_by_path[/datum/action/xeno_action/activable/ravage]
+		if(charge)
+			charge.clear_cooldown() //Reset charge cooldown
+		if(ravage)
+			ravage.clear_cooldown() //Reset ravage cooldown
+
+		for(var/mob/living/witness in hearers(rage_power_radius, X)) //Roar that applies cool SFX and some soft CC
+
+			if(!witness.hud_used)
+				continue
+
+			var/obj/screen/plane_master/floor/OT = witness.hud_used.plane_masters["[FLOOR_PLANE]"]
+			var/obj/screen/plane_master/game_world/GW = witness.hud_used.plane_masters["[GAME_PLANE]"]
+
+			addtimer(CALLBACK(OT, /atom.proc/remove_filter, "rage_outcry"), 1 SECONDS)
+			GW.add_filter("rage_outcry", 2, list("type" = "radial_blur", "size" = 0.07))
+			animate(GW.get_filter("rage_outcry"), size = 0.12, time = 5, loop = -1)
+			OT.add_filter("rage_outcry", 2, list("type" = "radial_blur", "size" = 0.07))
+			animate(OT.get_filter("rage_outcry"), size = 0.12, time = 5, loop = -1)
+			addtimer(CALLBACK(GW, /atom.proc/remove_filter, "rage_outcry"), 1 SECONDS)
+
+	for(var/turf/affected_tiles as() in RANGE_TURFS(rage_power_radius, X.loc))
+		affected_tiles.Shake(4, 4, 1 SECONDS) //SFX
+
+	for(var/mob/living/L in hearers(rage_power_radius, X)) //Roar that applies cool SFX and some soft CC
```

Comment:
Dont spam hearers() its terribad for performance

Reply:
You can cache the results of hearers() up in this file for the first time is used and reuse the results here.

It's still only 2 hearers() on a 2 minute cooldown so it can't be that expensive anyway.